controllers/etcdcluster_controller.go

```diff
@@ -473,10 +476,17 @@ func (r *EtcdClusterReconciler) reconcile(
 	return result, nil, nil
 }
 
-// nextOutdatedPeer returns an EtcdPeer which has a different version than the
-// EtcdCluster.
+type outdatedReason int
+
+const (
+	outdatedVersion outdatedReason = iota + 1
+	outdatedPodTemplate
```

Comment:
Is this using the defaulting to infer a type of `outdatedReason` and a value of `0`?

Reply:
Not sure I fully understand your question. This is just to have a constant to differentiate between the different outdated reasons. The enumeration starts with 1 on purpose so that a possible use of a default value (`0`) doesn't accidentally signal a specific reason where there is none.